src/app/services/data.service.ts

```diff
@@ -37,17 +37,22 @@ export class DataService {
       this.$schema
     ]).pipe(
       filter(([ds, schema]) => !!schema),
-      switchMap(([ds, schema]) => this.http.get<Aggregate>(ds.url))
+      switchMap(([ds, schema]) => this.http.get<Aggregate>(ds.url).pipe(
+        map(aggregate => ({ aggregate, error: null })),
+        catchError(err => of({ aggregate: {}, error: 'The selected data file could not be parsed. Please check your file format and try again.'}))
+      )),
     ).subscribe({
-      next: agg => {
+      next: ({ aggregate, error }) => {
         const validate = this.ajv.compile(this.$schema.value as JSONSchemaType<Aggregate>);
-        if (validate(agg)) {
-          this.$aggregate.next(agg);
+        if (error) {
+          this.addError(error);
+        } else if (validate(aggregate)) {
+          this.$aggregate.next(aggregate);
+          this.$errors.next([]);
         } else {
-          this.addError('Error parsing data file: ' + validate.errors![0].message + '. Please refresh, check your file format, and try again.');
+          this.addError('Error parsing data file: ' + validate.errors![0].message + '. Please check your file format and try again.');
```

Comment:
if this error string is simply dumped to the console, i think it's good. otherwise, consider adding two <br> line breaks (if html) to give it space in the interface to add emphasis to the action